test/commands/schema.test.js

```diff
@@ -0,0 +1,132 @@
+const fs = require("fs");
+const path = require("path");
+const { expect, test } = require("@oclif/test");
+const { query: q } = require("faunadb");
+const { withOpts, getEndpoint, matchFqlReq } = require("../helpers/utils.js");
+const { ux } = require("@oclif/core");
+
+const files = {
+  version: 0,
+  files: [
+    { filename: "main.fsl" },
+    { filename: "functions.fsl" },
+    { filename: "legacy.json" },
+  ],
+};
+
+const main = {
+  version: 0,
+  content: "collection foo {}\n\nfunction bar(x, y) { x + y }\n",
+};
+
+const functions = {
+  version: 0,
+  content: "function id(x) { x }",
+};
+
+const diff = {
+  version: 0,
+  diff: "main.fsl ADD collection foo",
+};
+
+const pullfiles = {
+  version: 0,
+  files: [{ filename: "main.fsl" }, { filename: "functions.fsl" }],
+};
+
+const updated = { version: 1 };
+
+describe("fauna schema ls test", () => {
+  test
+    .nock(getEndpoint(), { allowUnmocked: false }, (api) =>
+      api
+        .persist()
+        .post("/", matchFqlReq(q.Now()))
+        .reply(200, new Date())
+        .get("/schema/1/files")
+        .reply(200, files)
+    )
+    .stdout()
+    .command(withOpts(["schema ls"]))
+    .it("runs schema ls", (ctx) => {
+      expect(ctx.stdout).to.contain(
+        "Schema files:\n\nmain.fsl\nfunctions.fsl\nlegacy.json"
+      );
+    });
+});
+
+describe("fauna schema cat test", () => {
+  test
+    .nock(getEndpoint(), { allowUnmocked: false }, (api) =>
+      api
+        .persist()
+        .post("/", matchFqlReq(q.Now()))
+        .reply(200, new Date())
+        .get("/schema/1/files/main.fsl")
+        .reply(200, main)
+    )
+    .stdout()
+    .command(withOpts(["schema cat", "main.fsl"]))
+    .it("runs schema cat", (ctx) => {
+      expect(ctx.stdout).to.contain(`${main.content}`);
+    });
+});
+
+describe("fauna schema push test", () => {
+  test
+    .stub(ux, "confirm", () => async () => true)
+    .nock(getEndpoint(), { allowUnmocked: false }, (api) =>
+      api
+        .persist()
+        .post("/", matchFqlReq(q.Now()))
+        .reply(200, new Date())
+        .post("/schema/1/validate?force=true")
+        .reply(200, diff)
+        .post("/schema/1/update?version=0")
+        .reply(200, updated)
+    )
+    .stdout()
+    .command(withOpts(["schema push", "--dir=test/testdata"]))
+    .it("runs schema push", (ctx) => {
+      expect(ctx.stdout).to.contain(`${diff.diff}`);
+    });
+});
+
```

Comment:
would be good to test both this and push with some nested directory structure.

The --retain behavior needs testing as well.

Reply:
I added some more annoying javascript tests.